frame/contracts/src/weights.rs

```diff
@@ -150,1183 +150,1191 @@ pub trait WeightInfo {
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	fn on_initialize() -> Weight {
-		(3_659_000 as Weight)
+		(3_972_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 	}
 	fn on_initialize_per_trie_key(k: u32, ) -> Weight {
-		(40_731_000 as Weight)
-			// Standard Error: 4_000
-			.saturating_add((2_317_000 as Weight).saturating_mul(k as Weight))
+		(44_890_000 as Weight)
+			// Standard Error: 5_000
+			.saturating_add((2_325_000 as Weight).saturating_mul(k as Weight))
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(k as Weight)))
 	}
 	fn on_initialize_per_queue_item(q: u32, ) -> Weight {
-		(384_459_000 as Weight)
-			// Standard Error: 45_000
-			.saturating_add((146_401_000 as Weight).saturating_mul(q as Weight))
+		(268_333_000 as Weight)
+			// Standard Error: 32_000
+			.saturating_add((155_020_000 as Weight).saturating_mul(q as Weight))
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn update_schedule() -> Weight {
-		(27_803_000 as Weight)
+		(27_452_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
-	fn put_code(n: u32, ) -> Weight {
+	fn instantiate_with_code(c: u32, n: u32, s: u32, ) -> Weight {
 		(0 as Weight)
-			// Standard Error: 208_000
-			.saturating_add((110_774_000 as Weight).saturating_mul(n as Weight))
-			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+			// Standard Error: 95_000
+			.saturating_add((154_745_000 as Weight).saturating_mul(c as Weight))
+			// Standard Error: 47_000
+			.saturating_add((370_000 as Weight).saturating_mul(n as Weight))
+			// Standard Error: 47_000
+			.saturating_add((2_636_000 as Weight).saturating_mul(s as Weight))
+			.saturating_add(T::DbWeight::get().reads(6 as Weight))
+			.saturating_add(T::DbWeight::get().writes(5 as Weight))
 	}
 	fn instantiate(n: u32, s: u32, ) -> Weight {
-		(175_290_000 as Weight)
+		(189_707_000 as Weight)
 			// Standard Error: 1_000
-			.saturating_add((3_000 as Weight).saturating_mul(n as Weight))
+			.saturating_add((4_000 as Weight).saturating_mul(n as Weight))
 			// Standard Error: 1_000
-			.saturating_add((2_244_000 as Weight).saturating_mul(s as Weight))
+			.saturating_add((2_256_000 as Weight).saturating_mul(s as Weight))
 			.saturating_add(T::DbWeight::get().reads(6 as Weight))
-			.saturating_add(T::DbWeight::get().writes(3 as Weight))
+			.saturating_add(T::DbWeight::get().writes(4 as Weight))
```

Comment:
The additional write is for incrementing the refcount of the `PrefabWasmModule`. Previously, this storage item was only read in order to get the constructor.